### DISCUSSION TITLE
Default File Structure

### DISCUSSION THREAD

**Discussion by devurose** [godotengine/godot-proposals#6353] [Editor]
It's a simple ask, but it would be wonderful if you could create a sort of default assortment of directories that replace the existing ones when creating a new project, so a template project I'd need to duplicate isn't necessary if I want to have a sort of uniform and organized file structure.
Maybe throw it somewhere in the editor settings? You could have it reference a folder on your system and basically duplicate it to a fresh project. I'm not versed in how Godot handles default files, but I'm relatively confident this is possible to implement to a degree.

**Comment by dukemagus** [comment]
Each project is unique. It's easier that you create the proper structure and save that project to use as template whenever you need

**Reply by devurose** [reply]
I'm not sure the each project being unique tidbit is necessarily true, you can organize the same for different projects

**Comment by Maximinodotpy** [comment]
Sounds good, but I myself havent made a project where I was fully content with the structure. Do you have something in mind what such a structure could look like?

**Reply by devurose** [reply]
I quite enjoy this layout for my games, it's usually worked in most case scenarios with little tweaking
![File System](https://user-images.githubusercontent.com/87827260/224474748-77c34007-f8ca-4af9-9807-67a71f3c6706.png)

**Reply by Maximinodotpy** [reply]
Looks interesting, so in `Assets` there are the "raw" files and in `Resources` there are actual game things [.res, .tres]?

Or did I misunderstand it.

Nice Font BTW

**Reply by devurose** [reply]
100%, that's exactly how I use this structure. But of course, people might organize differently. Thus my pitch on a way to customize the default file structure :)

I know It'd save me some time on my workflow, that's for sure

**Reply by devurose** [reply]
Also thanks, I went on an editor deco spree and had a little fun with it haha

**Comment by vnen** [comment]
I think it should just properly support template projects. Right now you can create a project from asset library template but AFAIK you cannot use a local project as a template.
